Source/Core/DiscIO/VolumeVerifier.cpp

```diff
@@ -0,0 +1,648 @@
+
+    const FileSystem* fs = m_volume.GetFileSystem(partition);
+    if (!fs)
+      break;
+
+    biggest_offset = std::max(biggest_offset, m_volume.PartitionOffsetToRawOffset(
+                                                  GetBiggestUsedOffset(fs->GetRoot()), partition));
+  }
+  return biggest_offset;
+}
+
+u64 VolumeVerifier::GetBiggestUsedOffset(const FileInfo& file_info) const
+{
+  if (file_info.IsDirectory())
+  {
+    u64 biggest_offset = 0;
+    for (const FileInfo& f : file_info)
+      biggest_offset = std::max(biggest_offset, GetBiggestUsedOffset(f));
+    return biggest_offset;
+  }
+  else
+  {
+    return file_info.GetOffset() + file_info.GetSize();
+  }
+}
+
+void VolumeVerifier::CheckMisc()
+{
+  const Region region = m_volume.GetRegion();
+  const Platform platform = m_volume.GetVolumeType();
+  const std::string game_id_unencrypted = m_volume.GetGameID(PARTITION_NONE);
+  const std::string game_id_encrypted = m_volume.GetGameID(m_volume.GetGamePartition());
+
+  if (game_id_unencrypted != game_id_encrypted)
+  {
+    bool inconsistent_game_id = true;
+    if (game_id_encrypted == "RELSAB")
+    {
+      if (StringBeginsWith(game_id_unencrypted, "410"))
+      {
+        // This is the Wii Backup Disc (aka "pinkfish" disc),
+        // which legitimately has an inconsistent game ID.
+        inconsistent_game_id = false;
+      }
+      else if (StringBeginsWith(game_id_unencrypted, "010"))
+      {
+        // Hacked version of the Wii Backup Disc (aka "pinkfish" disc).
+        std::string proper_game_id = game_id_unencrypted;
+        proper_game_id[0] = '4';
+        AddProblem(Severity::Low,
+                   StringFromFormat(GetStringT("The game ID is %s but should be %s.").c_str(),
+                                    game_id_unencrypted.c_str(), proper_game_id.c_str()));
+        inconsistent_game_id = false;
+      }
+    }
+
+    if (inconsistent_game_id)
+    {
+      AddProblem(Severity::Low, GetStringT("The game ID is inconsistent."));
+    }
+  }
+
+  if (game_id_encrypted.size() < 4)
+  {
+    AddProblem(Severity::Low, GetStringT("The game ID is unusually short."));
+  }
+  else
+  {
+    const char country_code = game_id_encrypted[3];
+    if (CountryCodeToRegion(country_code, platform, region) != region)
+    {
+      AddProblem(
+          Severity::Medium,
+          GetStringT("The region code does not match the game ID. If this is because the "
+                     "region code has been modified, the game might run at the wrong speed, "
+                     "graphical elements might be offset, or the game might not run at all."));
+    }
+  }
+
+  const IOS::ES::TMDReader& tmd = m_volume.GetTMD(m_volume.GetGamePartition());
+  if (tmd.IsValid())
+  {
+    const u64 ios_id = tmd.GetIOSId() & 0xFF;
+
+    if (region == Region::NTSC_K && ios_id < 40 && ios_id != 4 && ios_id != 9 && ios_id != 21 &&
+        ios_id != 37)
```

Comment:
This IOS list looks a bit odd, I don't see a correlation between 37 and the others. I mean, IOS37 fixed the hash comparison, but why the others? Non-Korean? WiiBrew isn't particularly helping there; at least not the [IOS History](https://wiibrew.org/wiki/IOS_History)
Also, thoughts on putting those into an array and doing a contains check instead?

Reply:
I got the list from https://hackmii.com/2008/09/korean-wii/. Maybe I should add that link to a comment? (The other IOSes below 40 did get added to Korean Wiis later on by system updates, but since they were old at that point, I don't think any games used them.)

Since the checks are small enough to basically fit on one line, I think it would just be more cumbersome to put them in an array (or a set, rather).

Reply:
Ok, good to know. I'd like some sort of comment then so you can immediately see it is excluding Korean ones (which would be allowed here).